src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java

```diff
@@ -436,13 +439,16 @@ public void appendEnd(final StringBuffer buffer, final Object object) {
      * @param buffer  the <code>StringBuffer</code> to populate
      * @since 2.0
      */
+    @SuppressWarnings("index") /* for (@LTEqLengthOf({"len","sepLen"}) int i = 0; i < sepLen; i++) { // i < sepLen.length() and len >= sepLen => i < len.length()
+    if (buffer.charAt(len - 1 - i) != fieldSeparator.charAt(sepLen - 1 - i)) { // len - 1 - i and seplen - 1 - i is NonNegative within the loop
+    */
     protected void removeLastFieldSeparator(final StringBuffer buffer) {
         final int len = buffer.length();
         final int sepLen = fieldSeparator.length();
         if (len > 0 && sepLen > 0 && len >= sepLen) {
             boolean match = true;
-            for (int i = 0; i < sepLen; i++) {
-                if (buffer.charAt(len - 1 - i) != fieldSeparator.charAt(sepLen - 1 - i)) {
+            for (@NonNegative @LTEqLengthOf({"len","sepLen"}) int i = 0; i < sepLen; i++) { // i < sepLen.length() and len >= sepLen => i < len.length()
+                if (buffer.charAt(len - 1 - i) != fieldSeparator.charAt(sepLen - 1 - i)) { // len - 1 - i and seplen - 1 - i is NonNegative within the loop
```

Comment:
I commented on this in your last pull request: `len` and `sepLen` are not array-like structures, so they should not be the targets of an upperbound annotation.